### PR TITLE
Rhodanzi trainer school npcs

### PR DESCRIPTION
**Pull request by assassinsgreed**
Implementing additional Rhodanzi City school events

### CODE REVIEW

File: strings/scripts/rhodanzi_city/rhodanzi_trainer_school.string [hunk 1]
```diff
@@ -1,11 +1,53 @@
...
+[GREEN]Grass
+
+#org @gText_TypeFire
+[RED]Fire
+
+#org @gText_TypeWater
+[BLUE]Water
+
+#org @gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizComplete
+Good luck, trainer! Believe in\nyour Pok\emon and you'll do great.
```
Comment (CrystalAQuick) [inline strings/scripts/rhodanzi_city/rhodanzi_trainer_school.string:62]: new line

File: eventscripts [hunk 2]
```diff
@@ -21,4 +21,11 @@ npc 3 0 0 EventScript_AnthraTown_FlowerGirl
...
-npc 6 0 0 EventScript_RhodanziTrainerSchool_Receptionist
\ No newline at end of file
+## Trainer School main room
+npc 6 0 0 EventScript_RhodanziTrainerSchool_MainRoom_Receptionist
+npc 6 0 1 EventScript_RhodanziTrainerSchool_MainRoom_FossilStudent
+npc 6 0 2 EventScript_RhodanziTrainerSchool_MainRoom_Librarian
+npc 6 0 3 EventScript_RhodanziTrainerSchool_MainRoom_Professor
+
+## Trainer School basic course
+npc 6 1 0 EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent
```
Comment (CrystalAQuick) [inline eventscripts:31]: new line

File: assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s [hunk 3]
```diff
@@ -4,18 +4,98 @@
...
+    @TODO: Populate. Needs to call waitfanfare and give the right item, based on 0x4031
+    end
+
+EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizFail:
+    @TODO: Populate
+    end
+
+EventScript_TypeStudentQuizComplete:
+    msgbox gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizComplete MSG_FACE
+    end
```
Comment (CrystalAQuick) [inline assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s]: new line

**Review by CrystalAQuick — CHANGES_REQUESTED** [review]

File: strings/scripts/rhodanzi_city/rhodanzi_trainer_school.string [hunk 4]
```diff
@@ -1,11 +1,62 @@
...
+Great! This will be fun.
+
+#org @gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_GrassQuestion
+You chose a [GREEN]grass[BLUE] type starter.\nWhich type is it strong against?
+
+#org @gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_FireQuestion
+You chose a [RED]fire[BLUE] type starter.\nWhich type is it strong against?
+
+#org @gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_WaterQuestion
+You chose a [BLUE]water[BLUE] type starter.\nWhich type is it strong against?
```
Comment (assassinsgreed) [inline strings/scripts/rhodanzi_city/rhodanzi_trainer_school.string:41]: wtf man

File: assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s [hunk 5]
```diff
@@ -3,19 +3,141 @@
...
+    if TRUE _call EventScript_TypeStudentQuizComplete
+    applymovement 0x1 m_LookUp
+    waitmovement ALLEVENTS
     release
     end
+
+EventScript_TypeStudentQuiz:
+    msgbox gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizIntro MSG_YESNO
+    compare LASTRESULT YES
+    if NO _goto EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizRejected
```
Comment (assassinsgreed) [inline assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s:73]: No -> False

File: assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s [hunk 6]
```diff
@@ -3,19 +3,141 @@
...
+    if TRUE _call EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizQuestionGrass
+    compare 0x4031 1 @ Fire
+    if TRUE _call EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizQuestionFire
+    compare 0x4031 2 @ Water
+    if TRUE _call EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizQuestionWater
+    return
+
+EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizChoices:
+    multichoiceoption gText_TypeGrass 0
+	multichoiceoption gText_TypeFire 1
```
Comment (assassinsgreed) [inline assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s:89]: weird indentation

File: assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s [hunk 7]
```diff
@@ -3,19 +3,141 @@
...
+    if notequal _goto EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizFail
+    if equal _call EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizSuccess
+    obtainitem ITEM_CHARCOAL 0x1
+    msgbox gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_ItemInfo MSG_FACE
+    return
+
+EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizQuestionWater:
+    msgbox gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_WaterQuestion MSG_KEEPOPEN
+    call EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizChoices
+    compare LASTRESULT 0
```
Comment (assassinsgreed) [inline assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s:118]: wrong, should be 1

File: assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s [hunk 8]
```diff
@@ -3,19 +3,141 @@
...
+    return
+
+EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizFail:
+    msgbox gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizFail MSG_FACE
+    return
+
+EventScript_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizRejected:
+    msgbox gText_RhodanziTrainerSchool_BasicCourse_TypeStudent_QuizRejected MSG_FACE
+    applymovement 0x1 m_LookUp
+    waitmovement ALLEVENTS
```
Comment (assassinsgreed) [inline assembly/overworld_scripts/rhodanzi_city/rhodanzi_trainer_school.s:137]: use function

**Review by CrystalAQuick — APPROVED** [review]

**Pull request merged**